sklearn/cluster/_hdbscan/hdbscan.py

```diff
@@ -792,11 +793,11 @@ def fit(self, X, y=None):
             if self.algorithm == "brute":
                 mst_func = _hdbscan_brute
                 kwargs["copy"] = self.copy
-            elif self.algorithm == "kdtree":
+            elif self.algorithm == "kd_tree":
                 mst_func = _hdbscan_prims
                 kwargs["algo"] = "kd_tree"
                 kwargs["leaf_size"] = self.leaf_size
-            elif self.algorithm == "balltree":
+            elif self.algorithm == "ball_tree":
```

Comment:
Patch code coverage failed for this `line 800` due to missing else block at` elif == "ball_tree"`. Probably not added since parameter validation is completed by validate_parameter_constraints.